emulate/emulate_queue.py

```diff
@@ -163,21 +166,60 @@ def retrieve_next_animation() -> Code:
 		return code_from_priority_queue
 	return retrieve_random_animation()
 
+
 def reset_dmx_values():
 	number_of_values = 5 * 28 * 3  # 5 rows, 28 columns, 3 color values
 	serialized = ",".join("0" * number_of_values)
 	redis_db.set('DMXvalues', serialized)
 
 
+def start_udp_server():
+	UDPServerSocket = socket.socket(socket.AF_INET, type=socket.SOCK_DGRAM)
+	UDPServerSocket.setblocking(0)
+	UDPServerSocket.bind(("0.0.0.0", int(os.environ.get('UDP_SERVER_PORT', 20001))))
+	while Config.objects.first().udp_receive_run:
+		try:
+			bytesAddressPair = UDPServerSocket.recvfrom(2048)
+
+			message = bytesAddressPair[0]
+			message = message.decode('utf-8')
+			message = to_json(message)
+
+			if message and str(message.get("key")) == str(Config.objects.first().udp_key):
```

Comment:
Klucz udp z configa wyciągasz z bazy w każdej iteracji z pętli. Czyli x razy na sekunde jest tutaj zapytanie do bazy. Nie wiem czy jest cacheowane, wątpię.

Reply:
z tym że on się może w każdej chwili zmienić

Reply:
jak często zmieniasz klucz ?

Reply:
Troche to zoptymalizowalem